### PR TITLE
Many migrations improvements and bugfixes

### PR DESCRIPTION
**Pull request by ProGM**
This pull introduces/changes:
 * `drop_constraint` and `drop_index` now accepts `force: true`
 * `PendingMigrationError` now shows which migrations are pending
 * Fixed `silenced: true`, not working properly
 * Removed "strange" inheritance between `Neo4j::Migrations::Base` and the legacy `Neo4j::Migration` class
 * Avoid creating the `SchemaMigration` model constraint when it already exists

Pings:
@cheerfulstoic
@subvertallchris

### CODE REVIEW

File: lib/neo4j/migrations.rb [hunk 1]
```diff
@@ -11,7 +11,8 @@ module Migrations
     class << self
       def check_for_pending_migrations!
         runner = Neo4j::Migrations::Runner.new
-        fail ::Neo4j::PendingMigrationError if runner.pending_migrations?
+        pending = runner.pending_migrations
+        fail ::Neo4j::PendingMigrationError, pending if pending.any?
```
Comment (cheerfulstoic) [inline lib/neo4j/migrations.rb]: What does this output?  Perhaps `pending.inspect` to turn it into a string?

Comment (ProGM) [inline lib/neo4j/migrations.rb]: @cheerfulstoic check here:
https://github.com/neo4jrb/neo4j/pull/1309/files#diff-3ced7dc77d960f7c24fdee3fe17efe5cR40
It's passed as a parameter to the error, and joined with "\n"

**Comment by ProGM** [comment]
Merged!

File: lib/neo4j/migrations.rb (hunk 1 above)
Comment (cheerfulstoic) [inline lib/neo4j/migrations.rb]: Oh, nice ;)  Thanks for the clarification

**Pull request merged**